presto-main/src/main/java/com/facebook/presto/operator/aggregation/state/StateCompiler.java

```diff
@@ -669,19 +658,53 @@ else if (method.getName().startsWith("set")) {
         private final String getterName;
         private final Class<?> type;
         private final Object initialValue;
+        private final Type sqlType;
 
         private StateField(String name, Class<?> type, Object initialValue)
         {
             this(name, type, initialValue, "get" + name);
         }
 
+        private StateField(String name, Class<?> type, Object initialValue, Type spiType)
+        {
+            this(name, type, initialValue, "get" + name, spiType);
+        }
+
         private StateField(String name, Class<?> type, Object initialValue, String getterName)
         {
+            this(name, type, initialValue, getterName, sqlTypeFromStackType(type));
+        }
+
+        private StateField(String name, Class<?> type, Object initialValue, String getterName, Type sqlType)
+        {
             this.name = requireNonNull(name, "name is null");
             checkArgument(!name.isEmpty(), "name is empty");
             this.type = requireNonNull(type, "type is null");
             this.getterName = requireNonNull(getterName, "getterName is null");
             this.initialValue = initialValue;
+            this.sqlType = requireNonNull(sqlType, "sqlType is null");
+        }
+
+        private static Type sqlTypeFromStackType(Class<?> stackType) {
+            if (stackType == long.class) {
+                return BIGINT;
+            }
+            else if (stackType == double.class) {
```

Comment:
All of these `else` clauses are redundant (since the `if` block returns). There's an IDEA inspection for this.